python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -245,6 +254,8 @@ def post(self, mine_no_or_guid=None):
         mine_region = data['mine_region']
         mine = Mine(mine_guid=uuid.uuid4(), **self.get_create_update_dict())
         try:
+            # query the mine tables and check if that mine name exists
+            self.throw_error_if_mine_exists( data['name'])
```

Comment:
Let's add `mine_name = data['name']` above this line and then use that variable here & on line 262